src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/StorageWatcher.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Diagnostics.DataProviders;
+using Diagnostics.RuntimeHost.Models;
+using Diagnostics.RuntimeHost.Services.StorageService;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Diagnostics.ModelsAndUtils.Models.Storage;
+using Newtonsoft.Json;
+using Diagnostics.RuntimeHost.Utilities;
+using System.IO;
+using Diagnostics.Logger;
+
+namespace Diagnostics.RuntimeHost.Services.SourceWatcher.Watchers
+{
+    public class StorageWatcher : ISourceWatcher
+    {
+        private IStorageService storageService;
+        private IHostingEnvironment hostingEnvironment;
+        private IConfiguration configuration;
+        private IGithubClient gitHubClient;
+
+        public StorageWatcher(IHostingEnvironment env, IConfiguration config, IStorageService service)
+        {
+            storageService = service;
+            hostingEnvironment = env;
+            configuration = config;
+            gitHubClient = new GithubClient(env, config);
+        }
+        public async Task<HealthCheckResult> CheckHealthAsync(CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task CreateOrUpdatePackage(Package pkg)
+        {
+            if (pkg == null)
+            {
+                throw new ArgumentNullException(nameof(pkg));
+            }
+            try
+            {
+                var blobName = $"{pkg.Id.ToLower()}/{pkg.Id.ToLower()}.dll";
+                var etag = await storageService.LoadBlobToContainer(blobName, pkg.DllBytes);
+                if (string.IsNullOrWhiteSpace(etag))
+                {
+                    DiagnosticsETWProvider.Instance.LogAzureStorageMessage(nameof(StorageWatcher), $"Uploading {pkg.Id} to blob failed, not proceeding further");
+                    return;
```

Comment:
should we throw a specific exception here instead of returning success?

Reply:
Since we are still in beta mode for this, I thought it would be okay to just log the exception than fail the API by throwing exception

Reply:
Oh I see.. this gets called along with GitHub package update.
Then before switching to the new source watcher completely Rekha, we would have to run your function app to make sure that blobs are up-to date.

Reply:
Yes, we can do that